src/coreclr/System.Private.CoreLib/src/System/StubHelpers.cs

```diff
@@ -103,13 +104,19 @@ internal static unsafe IntPtr ConvertToNative(int flags, string strManaged, IntP
                     // wasting memory on systems with multibyte character sets where the buffer we end up with is often much
                     // smaller than the upper bound for the given managed string.
 
-                    byte[] bytes = AnsiCharMarshaler.DoAnsiConversion(strManaged,
-                        fBestFit: 0 != (flags & 0xFF), fThrowOnUnmappableChar: 0 != (flags >> 8), out nb);
+                    byte[] bytes = ArrayPool<byte>.Shared.Rent(checked((strManaged.Length + 1) * Marshal.SystemMaxDBCSCharSize));
+                    fixed (byte* bytesPtr = &bytes[0])
+                    {
+                        bool fBestFit = 0 != (flags & 0xFF);
+                        bool fThrowOnUnmappableChar = 0 != (flags >> 8);
+                        nb = Marshal.StringToAnsiString(strManaged, bytesPtr, bytes.Length, fBestFit, fThrowOnUnmappableChar);
```

Comment:
This represents a serious issue. If `fThrowOnUnmappableChar` is `true`, then this can throw. This then means that the rented `ArrayPool` array will "leak" since it can never be reused.

Reply:
Thank you for the review. 

In throw case, the rented array will be collected as there is no reference to it.
`ArrayPool.Shared.Rent()` could reassign an array when the same slot size is requested.

Reply:
Oh, I didn't realize that not returning the array was okay. Thanks.